addons/uom/models/uom_uom.py

```diff
@@ -62,49 +113,31 @@ def _onchange_uom_type(self):
         if self.uom_type == 'reference':
             self.factor = 1
 
-    @api.constrains('category_id', 'uom_type', 'active')
-    def _check_category_reference_uniqueness(self):
```

Comment:
We couldn't keep the SQL constraint? It's faster.

Reply:
sql constraint makes impossible to change the reference unit: work on each insert/update to check there is one and only one.
changing the reference unit needs coordinated update of 2 records and should not occurs that often